lib/omnibus/software.rb

```diff
@@ -655,7 +655,7 @@ def with_standard_compiler_flags(env = {}, opts = {})
         "LD_RUN_PATH" => "#{install_dir}/embedded/lib"
       }
 
-      if solaris2?
+      if solaris2? && Ohai['os_version'].to_i <= 5.10
```

Comment:
We should use an actual version constraint comparison here, something like this should work:

```ruby
if solaris2? && platform_version.satisfies?('<= 5.10')
```